### ISSUE TITLE
Disable buttons when selected port is blank ""

### ISSUE THREAD

**Issue by PropGit** [parallaxinc/solo#438]
**Background:**

BlocklyProp Launcher v1.0.1 adds enhanced wired and wireless port management to ease the user experience.  Part of this includes important new behavior when a selected port becomes disconnected (as is common with USB robots) - the new port list includes a blank "" (empty string) port as the first port.  This causes Solo to leave the _selected_ port blank.  This is fully intended; we don't want to auto-select another port at that point, but rather expect the old port to return sometime later.

**Needed Enhancement:**
When the selected item in the port list is an empty string "", disable the Download RAM, Download EEPROM, Open Terminal, and Open Graph buttons, to 1) prevent unsuccessful download/terminal/graph connection events, and 2) to alert the user to the need to either plug in the board again or select a different port.

Currently, Solo does this:
![image](https://user-images.githubusercontent.com/1266318/82612012-b5fe2780-9b76-11ea-98ae-df6af7578659.png)

FYI: here's a view of the user starting to select a different port (the blank port is still listed here):

![image](https://user-images.githubusercontent.com/1266318/82612061-d5955000-9b76-11ea-8188-4eaa27d292be.png)

**Comment by zfi** [comment]
There is an update in the feature-modules branch that I believe addresses this concern. When the UI is initialized, the default state of the port drop down contains the string 'Searching...". When the user clicks on the drop down, the Searching string is selected but nothing come of it. 
<img width="280" alt="Screen Shot 2020-05-21 at 3 41 33 PM" src="https://user-images.githubusercontent.com/8442260/82613221-a03e3180-9b79-11ea-8e1d-b892ff62cd1f.png">
<img width="280" alt="Screen Shot 2020-05-21 at 3 41 40 PM" src="https://user-images.githubusercontent.com/8442260/82613246-b0eea780-9b79-11ea-93a3-b2245c60222e.png">

Once a Solo receives a populated port list, it displays the ports in the order in which they were received. The action buttons state is also updated based on the connection state as demonstrated in the images below. 

<img width="360" alt="Screen Shot 2020-05-21 at 3 45 47 PM" src="https://user-images.githubusercontent.com/8442260/82613608-8d782c80-9b7a-11ea-9433-d214252da744.png">
<img width="360" alt="Screen Shot 2020-05-21 at 3 45 42 PM" src="https://user-images.githubusercontent.com/8442260/82613620-95d06780-9b7a-11ea-8b20-45e1382b1279.png">

**Comment by PropGit** [comment]
That looks excellent and is very worthy and needed; however, I think this is a different case.

In the "Searching..." case, the port list is empty (no port strings at all)... but in the case I'm talking about, the port list *is not* empty and contains a empty string "" as it's first item, with the next item(s) containing visible text.  That's intentional; however, when the first item is selected (as it does naturally; and I want that behavior to stay) there's no point in trying to download or open a terminal or graph because there's no port described by that empty selection... it's just a placeholder for a port which may return later when reconnected.

Currently, I can download with this blank port selected and Launcher returns an error that it can not find the port "null".  I can make that message friendlier, but we should actually prevent the situation in the first place as it will nicely trigger the user to notice that they have no port selected... and that must be why the download buttons are disabled. :-)

**Comment by zfi** [comment]
I was unaware that an empty string in the port list was possible. What is the purpose of an empty port value? Solo as it's now coded has no idea what to do with that.

**Comment by PropGit** [comment]
It wasn't possible before, at least Launcher never emitted an empty port
name before - it's a new feature I added today to solve a problem.

The problem... I wanted our system to behave better when ports are arriving
and departing- it's a constant problem for customers. It occurred to me
that a technique I used in the BASIC Stamp Editor and Parallax Serial
Terminal would work here where, when a port disappears, using a placeholder
(empty port name, in this case) keeps the system from automatically
selecting a different port when the recently departed is actually likely to
return. Works beautifully with the existing system!  I tried it, it worked,
I ran with it and shored it up as a new feature.  No changes needed in Solo
for it to work, other than it should disable downloads and terminals when
the empty port name is selected.
